flink-core/src/main/java/org/apache/flink/util/ExceptionUtils.java

```diff
@@ -132,31 +132,39 @@ public static boolean isJvmFatalOrOutOfMemoryError(Throwable t) {
 	}
 
 	/**
-	 * Generates new {@link OutOfMemoryError} with more detailed message.
+	 * Tries to enrich the passed exception with additional information.
 	 *
 	 * <p>This method improves error message for direct and metaspace {@link OutOfMemoryError}.
 	 * It adds description of possible causes and ways of resolution.
 	 *
 	 * @param exception The exception to enrich.
 	 * @return either enriched exception if needed or the original one.
 	 */
-	public static Throwable enrichTaskManagerOutOfMemoryError(Throwable exception) {
-		if (isMetaspaceOutOfMemoryError(exception)) {
-			return changeOutOfMemoryErrorMessage(exception, TM_METASPACE_OOM_ERROR_MESSAGE);
-		} else if (isDirectOutOfMemoryError(exception)) {
-			return changeOutOfMemoryErrorMessage(exception, TM_DIRECT_OOM_ERROR_MESSAGE);
+	public static Throwable tryEnrichTaskManagerError(Throwable exception) {
```

Comment:
Maybe add `@Nullable`

Reply:
Good point. I will add it. First I thought I would like to avoid encouraging the usage of this method with `null` values but actually it is not a problem.